File: Python3/2554. Maximum Number of Integers to Choose From a Range I.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        banned = set(banned)
        ans = 0

        for i in range(1, n + 1):
            if i not in banned and maxSum - i >= 0:
                ans += 1
                maxSum -= i

            if maxSum <= i:
                break
        return ans

File: Java/2554. Maximum Number of Integers to Choose From a Range I.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        Set<Integer> nums = new HashSet();
        for (int b: banned) {
            nums.add(b);
        }

        int ans = 0;
        for (int i = 1; i < n + 1; i++) {
            if (!nums.contains(i) && maxSum - i >= 0) {
                maxSum -= i;
                ans++;
            }
            if (maxSum <= i) {
                break;
            }
        }

        return ans;
    }
}